language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/helpers/TransactionsServiceImpl.scala

```diff
@@ -156,49 +140,38 @@ object TransactionServiceImpl {
     case Empty => ""
   }
 
-  def createWithRef(ledgerContent: Observable[LedgerItem])(
-      implicit ec: ExecutionContext): (ServerServiceDefinition, TransactionServiceImpl) = {
-    val serviceImpl = new TransactionServiceImpl(ledgerContent)
-    (TransactionServiceGrpc.bindService(serviceImpl, ec), serviceImpl)
+  def createWithRef(ledgerContent: Observable[LedgerItem], authorizer: Authorizer)(
+      implicit ec: ExecutionContext): (ServerServiceDefinition, TransactionsServiceImpl) = {
+    val impl = new TransactionsServiceImpl(ledgerContent)
+    val authImpl = new TransactionServiceAuthorization(impl, authorizer)
+    (TransactionServiceGrpc.bindService(authImpl, ec), impl)
   }
 
-  val ledgerOffsetOrdering: Ordering[LedgerOffset] = new Ordering[LedgerOffset] {
-    override def compare(x: LedgerOffset, y: LedgerOffset): Int = {
-      if (x.equals(y)) 0
-      else {
-        x.getAbsolute match {
-          case "" =>
-            x.getBoundary match {
-              case LEDGER_BEGIN => -1
-              case LEDGER_END => 1
-              case Unrecognized(value) =>
-                throw new RuntimeException(
-                  s"Found boundary that is neither BEGIN or END (value: $value)")
-            }
-          case xAbs =>
-            y.getAbsolute match {
-              case "" =>
-                y.getBoundary match {
-                  case LEDGER_BEGIN => 1
-                  case LEDGER_END => -1
-                  case Unrecognized(value) =>
-                    throw new RuntimeException(
-                      s"Found boundary that is neither BEGIN or END (value: $value)")
-                }
-              case yAbs => xAbs.compareTo(yAbs)
-            }
-        }
+  val ledgerOffsetOrdering: Ordering[LedgerOffset] = (x: LedgerOffset, y: LedgerOffset) => {
```

Comment:
We may want to add tests for this as the below logic does not seem to be trivial (I understand you just reformatted this method).

Reply:
Sure, can you open a ticket for this? This is beyond the scope of the work done in this PR.

Reply:
Created #3742 for reusing code for ledger offset ordering as we already have such logic littered around in our codebase.